src/ipc/ipc4/helper.c

```diff
@@ -582,7 +586,10 @@ int ipc4_create_chain_dma(struct ipc *ipc, struct ipc4_chain_dma *cdma)
 
 	memset_s(&params, sizeof(params), 0, sizeof(params));
 	memset_s(&copier_cfg, sizeof(copier_cfg), 0, sizeof(copier_cfg));
-	construct_config(&copier_cfg, cdma->data.r.fifo_size, &params, cdma->header.r.scs);
+	ret = construct_config(&copier_cfg, cdma->data.r.fifo_size, &params, cdma->header.r.scs);
+	if (ret < 0)
+		return IPC4_FAILURE;
```

Comment:
this should be
```
if (ret != IPC4_SUCCESS)
	return ret;
```
Yes, everywhere above too.

Reply:
Wow, there's a lot of them. I'll add a try to fix them to 2nd commit. Please note that I'm not able to test this.

Reply:
OK I did it, please check carefully the second commit. Many of the called functions are common with IPC3 so in them the return value need to be changed in this module. I changed those that did not seem to have IPC3 usage. @RanderWang Are you able to test this change? I have no idea if CI runs this.